### DISCUSSION TITLE
search_selection (*) should leave select mode

### DISCUSSION THREAD

**Discussion by chtenb** [helix-editor/helix#10246] [General]
It makes sense to have `search_selection` (bound to `*` by default) always leave select mode, since you generally want to search after that (by pressing n), and it's suprising/confusing when select mode is still on and pressing n causes multiple selections.

You can add this keybind manually, but I think it should rather be the default:

```
[keys.select]
"*" = [ "search_selection", "normal_mode" ]
```

~I guess the same argument could be made for the commands `search_next`, `search_prev`, `search`, and many other commands that tend to involve jumping around with the cursor.~

**Comment by TornaxO7** [comment]
You can turn this off, by remapping your `*` key if you don't want it.

There's a use-case for it's current behaviour as you can see in `helix --tutor` (or `hx --tutor`) and jump to section `9.2`.
```
=================================================================
=           9.2 ADDING SELECTION ON NEXT SEARCH MATCH           =
=================================================================

 A property of Select mode (v) when using n and N is that instead
 of moving the selection to the next match, it adds a new
 selection on each match.

 1. Move the cursor to the line marked '-->' below.
 2. Select the first "bat" and type * to set it to search.
 3. Type v to enter Select mode.
 4. Type n to select the other "bat".
 5. Use c or r to change the "bat"s to "cat".

 --> Everybody wants to be a bat,
 --> because a cat's the only bat
 --> who knows where it's at.
```

**Reply by chtenb** [reply]
Cool, I did not know this, and had not noticed either, because my hits were far out from each other. I will remove the `search_next` command et al from the proposal. I think the point still stands for `search_selection`? Since you use visual mode for selecting the initial text, which has nothing to do with whether you want multiple selections or not. I feel that should be a conscious decision by re-entering select mode after having pressed `*`.
